R/interop-parsnip.R

```diff
@@ -0,0 +1,68 @@
+#' Register mmrm for use with tidymodels
+#'
+#' @inheritParams base::requireNamespace
+#' @return A logical value indicating whether registration was successful
+#'
+#' @keywords internal
+parsnip_add_mmrm <- function(quietly = FALSE) {
+  if (!requireNamespace("parsnip", quietly = quietly)) return(FALSE)
+
+  parsnip::set_model_engine(
+    model = "linear_reg",
+    eng = "mmrm",
+    mode = "regression"
+  )
+
+  parsnip::set_dependency(
+    pkg = "mmrm",
+    model = "linear_reg",
+    eng = "mmrm",
+    mode = "regression"
+  )
+
+  parsnip::set_encoding(
+    model = "linear_reg",
+    eng = "mmrm",
+    mode = "regression",
+    options = list(
+      predictor_indicators = "none",
+      compute_intercept = FALSE,
+      remove_intercept = FALSE,
+      allow_sparse_x = FALSE
+    )
+  )
+
+  args <- fill_names(c(
+    "reml",
+    "n_cores"
+  ))
+
+  mapply(
+    parsnip = names(args),
+    original = args,
+    parsnip::set_model_arg,
```

Comment:
It's not clear to me how arguments registered with `parsnip::set_model_arg` are supposed to be used. From what I can tell, unless the argument maps to some `tidymodels` whitelist, it isn't used?

Maybe someone with more `tidymodels` familiarity can weigh in

Reply:
@yonicd do you know? (I'm so far not a user of that)

Reply:
I'll continue to dig into it today. From what I understand, `tidymodels` compartmentalizes arguments into tuning parameters and model behavior arguments.

I don't think we have anything that would be considered a tuning parameters, but I would like to expose the behavior arguments if possible.

Reply:
Welp, turns out you don't have to register args. It just wasn't clear to me how they get introduced. After some toying around with `tidymodels` today I found that by _not_ registering arguments, they can be passed to `set_engine` :shrug: 

This will also capture `...` arguments and propagate them to `control`/`h_get_optimizers`

```r
linear_reg() %>%
  set_engine("mmrm", reml = FALSE) %>%
  fit(FEV1 ~ RACE + ARMCD * AVISIT + us(AVISIT | USUBJID), fev_data)
```